### ISSUE TITLE
Include all hpHosts lists

### ISSUE THREAD

**Issue by dag** [gorhill/uMatrix#525]
Not necessarily all enabled by default. 

```
# Complete list
http://hosts-file.net/download/hosts.txt

# Additions since last release
http://hosts-file.net/hphosts-partial.txt

# ATS - Ad/Tracking Servers
http://hosts-file.net/ad_servers.txt

# EMD - Engaged in Malware Distribution
http://hosts-file.net/emd.txt

# EXP - Exploits
http://hosts-file.net/exp.txt

# FSA - Fraudulent Services and Applications
http://hosts-file.net/fsa.txt

# GRM - Grass Roots Marketing (astroturfing)
http://hosts-file.net/grm.txt

# HFS - hpHosts Forum Spammers
http://hosts-file.net/hfs.txt

# HJK - Hijacking
http://hosts-file.net/hjk.txt

# MMT - Misleading Marketing Tactics
http://hosts-file.net/mmt.txt

# PHA - Pharmacy (illegal)
http://hosts-file.net/pha.txt

# PSH - Phishing
http://hosts-file.net/psh.txt

# WRZ - Warez
http://hosts-file.net/wrz.txt
```

**Comment by N4melessGitter** [comment]
lol, why would anyone want to block Warez/Torrent sites? xD

Thanks for the list though, I will definitely use some of them.

**Comment by N4melessGitter** [comment]
Ok, on a more serious note - please don't add those lists @gorhill!
I only added 2 of the lists and it nearly trippled the blocked domains in total.
Just glancing over some of them, there were  f.e. dozens of subdomains for asstraffic18.com.
I don't understand why they wouldn't just block the whole damn domain instead.
@OP if you want those lists just add them to your own list, but I don't think it's necessary.
If you don't allow scripts even for 1st party, and considering you'll probably never come across 90+% of the domains in those lists, there's zero risk from not having those domains in the list, and it's just not worth the performance impact, no matter how small.

**Comment by dag** [comment]
I do have them added myself, and I agree most of these probably should not be enabled by default.  But I thought it could be nice to have them available as standard options, particularly combined with #526 since custom added lists don't have any names or descriptions.  It would also be nice if it listed the lists listing the blocked host when it blocks a host (nice sentence there) so if you visit a bittorrent site you get a message saying "blocked by these lists: WRZ - Warez" and you're better informed to add an exception if you want.

Is there any performance impact to having lists available but not enabled?  I can only imagine it as a UI issue, having too many lists available, but uMatrix could group lists like uBlock does and there could be a group "hpHosts classifications" or something.

>  I don't understand why they wouldn't just block the whole damn domain instead.

I don't think hosts files normally work like that.  I think it might be a special case of blocking tools like uMatrix.  The hosts file is not originally meant for blocking, but rather a static DNS lookup table.  So if you just route `127.0.0.1 example.com` in a hosts file that has no effect on `www.example.com`.

> If you don't allow scripts even for 1st party, and considering you'll probably never come across 90+% of the domains in those lists, there's zero risk from not having those domains in the list, and it's just not worth the performance impact, no matter how small.

I prefer to allow 1st-party scripts, but I do block everything 3rd-party by default.  Technically I could do completely without hosts lists but I like seeing which third-parties are more questionable to whitelist and I like being able quickly allow 3rd-party css and images while still blocking ads and such.  I also like that hosts files block even 1st-party requests in case I'm baited to click a link to a malware site or whatever.  I like being informed, and the hpHosts classifications give me more granularity.

**Comment by N4melessGitter** [comment]
> But I thought it could be nice to have them available as standard options, ...
I think that would just encourage people to enable those lists to be "more secure", while not allowing 1st-party scripts is simply the better and more secure solution with the best performance aswell.

> .. particularly combined with #526 
I think you made some good points there, and while I personally trust that the people who maintain those lists do a good job and I don't care which list blocks a certain domain because they get added for a reason, I certainly wouldn't mind it, if it can be done without much of a performance impact in a way that you suggested.

> if you visit a bittorrent site you get a message saying "blocked by these lists: WRZ - Warez"
that would never happen to me, so I can't comment on that any further ;-)

> Is there any performance impact to having lists available but not enabled?
I don't think so, but I wrote above why I'm against it.

> I prefer to allow 1st-party scripts
Ok, didn't expect that since I assumed most "advanced users" for whom uMatrix is intended for, had previously or are still using addons like noScript.
Most websites work just fine or at least good enough without scripts enabled.
With your setting I can understand that you may want to add as many blocklists as possible.
But you'll sacrifice safety for some more information, and will never be as safe as having a deny-all/allow-where-absolutely-necessary approach. Even with millions of domains blocked.


Since uMatrix is only recommended for advanced users anyway, I think every user could find more lists for his/her liking in no time if they deem it necessary.

Would you mind sharing how many blocked hostnames you currently have?

**Comment by N4melessGitter** [comment]
> But I thought it could be nice to have them available as standard options, ... <
I think that would just encourage people to enable those lists to be "more secure", while not allowing 1st-party scripts is simply the better and more secure solution with the best performance aswell.

> .. particularly combined with #526 <
I think you made some good points there, and while I personally trust that the people who maintain those lists do a good job and I don't care which list blocks a certain domain because they get added for a reason, I certainly wouldn't mind it, if it can be done without much of a performance impact in a way that you suggested.

> if you visit a bittorrent site you get a message saying "blocked by these lists: WRZ - Warez" <
that would never happen to me, so I can't comment on that any further ;-)

> Is there any performance impact to having lists available but not enabled? <
I don't think so, but I wrote above why I'm against it.

> I prefer to allow 1st-party scripts <
Ok, didn't expect that since I assumed most "advanced users" for whom uMatrix is intended for, had previously or are still using addons like noScript.
Most websites work just fine or at least good enough without scripts enabled.
With your setting I can understand that you may want to add as many blocklists as possible.
But you'll sacrifice safety for some more information, and will never be as safe as having a deny-all/allow-where-absolutely-necessary approach. Even with millions of domains blocked.


Since uMatrix is only recommended for advanced users anyway, I think every user could find more lists for his/her liking in no time if they deem it necessary.

Would you mind sharing how many blocked hostnames you currently have?

**Comment by N4melessGitter** [comment]
> But I thought it could be nice to have them available as standard options, ...

I think that would just encourage people to enable those lists to be "more secure", while not allowing 1st-party scripts is simply the better and more secure solution with the best performance aswell.

> ... particularly combined with #526

I think you made some good points there, and while I personally trust that the people who maintain those lists do a good job and I don't care which list blocks a certain domain because they get added for a reason, I certainly wouldn't mind it, if it can be done without much of a performance impact in a way that you suggested.

> if you visit a bittorrent site you get a message saying "blocked by these lists: WRZ - Warez"

that would never happen to me, so I can't comment on that any further ;-)

> Is there any performance impact to having lists available but not enabled?

I don't think so, but I wrote above why I'm against it.

> I prefer to allow 1st-party scripts

Ok, didn't expect that since I assumed most "advanced users" for whom uMatrix is intended for, had previously or are still using addons like noScript.
Most websites work just fine or at least good enough without scripts enabled.
With your setting I can understand that you may want to add as many blocklists as possible.
But you'll sacrifice safety for some more information, and will never be as safe as having a deny-all/allow-where-absolutely-necessary approach. Even with millions of domains blocked.


Since uMatrix is only recommended for advanced users anyway, I think every user could find more lists for his/her liking in no time if they deem it necessary.

Would you mind sharing how many blocked hostnames you currently have?

**Comment by dag** [comment]
> Would you mind sharing how many blocked hostnames you currently have?

```
451,513 distinct blocked hostnames
```

But I'm in an experimental stage of figuring out the setup that best suits me, so tomorrow that number might be zero or I might be using uBlock<sub>0</sub> instead etc etc.  ;-)  Currently it's mostly for the fun of seeing how many third-parties sites try to connect with that are blacklisted in ANY hosts file I can find.  I guess that's also in part why I'm interested in being able to see which list the blocking originates from: are the third-parties all just ads and trackers, or ever something potentially more malicious?

So, thinking this through more maybe I agree with closing this ticket.  My use case is highly specialized.  Still, @gorhill, did you consider adding ANY of these lists?  Some of them MIGHT arguably still be useful, considering you already include other malware lists for example.

Anyway, thanks for two verily awesome tools.
